lib/simplecov/filter.rb

```diff
@@ -63,11 +65,23 @@ def matches?(source_file)
   end
 
   class ArrayFilter < SimpleCov::Filter
-    # Returns true if any of the file paths passed in the given array matches the string
-    # configured when initializing this Filter with StringFilter.new(['some/path', 'other/path'])
+    def initialize(filter_argument)
+      filter_argument.map! do |arg|
+        if arg.is_a?(SimpleCov::Filter)
+          arg
+        else
+          Filter.class_for_argument(arg).new(arg)
```

Comment:
just thinking out loud, couldn't we also put that check into `Filter.class_for_argument` e.g. if it is already a filter? Would make that code easier and centralize the knowledge :)